### PR TITLE
Sof 2373/able to create modifier

### PR DESCRIPTION
**Pull request by LindvedKrvang**
Note: Dependent on https://github.com/tv2/sofie-web-client/pull/301

Able to create a PanelInputModifier.

There is now a "create" button that, when clicked, opens our Sidebar with a form for a PanelInputModifier. 
Filling out the form and clicking save will create a new Modifier.

### CODE REVIEW

File: src/app/settings/panel-settings/components/edit-panel-input-modifier/edit-panel-input-modifier.component.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  public cancel(): void {
+    this.onCancel.emit()
+  }
+
+  private updateSaveButtonTooltip(): void {
+    const invalidControlsLabels: string[] = this.findInvalidControlsLabels()
+    if (invalidControlsLabels.length > 0) {
+      this.saveButtonTooltip = $localize`common.form-required-field.button-tooltip` + ` ${invalidControlsLabels.join(', ')}`
```
Comment (RasmusAlbrektsen) [inline src/app/settings/panel-settings/components/edit-panel-input-modifier/edit-panel-input-modifier.component.ts:48]: This is very minor and i should have caught it when it came to. But when we use the `common.form-required-field.button-tooltip` we end with a period and then write the fields after. 
image

Maybe we should end with a `:` or write something following the statement to say e.g. (using the image example):
`The fields are: Model, Hostname`

Again, a minor thing, but it just stood out to me.

**Review by RasmusAlbrektsen — APPROVED** [review]
Nice work :-)

File: src/app/settings/panel-settings/components/edit-panel-configuration/edit-panel-configuration.component.ts [hunk 2]
```diff
@@ -54,6 +56,9 @@ export class EditPanelConfigurationComponent implements OnInit {
     if (this.panelConfiguration?.model) {
       this.updatePanelLayoutConfigurationOptions(this.panelConfiguration.model)
     }
+
+    this.updateSaveButtonTooltip()
+    this.form.valueChanges.subscribe(() => this.updateSaveButtonTooltip())
   }
 
   public save(): void {
```
Comment (MadsThy) [inline src/app/settings/panel-settings/components/edit-panel-configuration/edit-panel-configuration.component.ts:64]: I think you need a check in the save function to make sure it doesn't run if the button is disabled :)

```
  if(!form.valid){
    return
  }
```

File: src/app/settings/panel-settings/components/edit-panel-input-modifier/edit-panel-input-modifier.component.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      id: this.formBuilder.control(this.panelInputModifier?.id),
+      name: this.formBuilder.control(this.panelInputModifier?.name, Validators.required),
+    })
+
+    this.updateSaveButtonTooltip()
+    this.form.valueChanges.subscribe(() => this.updateSaveButtonTooltip())
+  }
+
+  public save(): void {
+    this.onSave.emit(this.form.value)
```
Comment (MadsThy) [inline src/app/settings/panel-settings/components/edit-panel-input-modifier/edit-panel-input-modifier.component.ts:38]:
```suggestion
    if(!form.valid) {
     return
    }
    this.onSave.emit(this.form.value)
```

File: src/app/settings/panel-settings/components/edit-panel-input-modifier/edit-panel-input-modifier.component.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  templateUrl: './edit-panel-input-modifier.component.html',
+})
+export class EditPanelInputModifierComponent implements OnInit {
+  @Input()
+  public panelInputModifier?: PanelInputModifier
+
+  @Output()
+  public onSave: EventEmitter<PanelInputModifier> = new EventEmitter()
+  @Output()
+  public onCancel: EventEmitter<void> = new EventEmitter()
```
Comment (MadsThy) [inline src/app/settings/panel-settings/components/edit-panel-input-modifier/edit-panel-input-modifier.component.ts:16]:
```suggestion
  @Input() public panelInputModifier?: PanelInputModifier

  @Output() public onSave: EventEmitter<PanelInputModifier> = new EventEmitter()
  @Output() public onCancel: EventEmitter<void> = new EventEmitter()
```

**Review by MadsThy — APPROVED** [review]

File: src/app/settings/panel-settings/components/edit-panel-configuration/edit-panel-configuration.component.ts (hunk 2 above)
Comment (LindvedKrvang) [reply src/app/settings/panel-settings/components/edit-panel-configuration/edit-panel-configuration.component.ts:64]: The button is outside of the `save()` function scope. So the `save()` function should not know anything about a button.
This seems like a minor detail, but it is quite an important distinction (that appliess to all functions).

That being said: The `form` is not part of the button and the `form` is something the `save()` function does know about and it wouldn't be a bad idea for the `save()` function to validate that the `form` is valid before saving :)

File: src/app/settings/panel-settings/components/edit-panel-input-modifier/edit-panel-input-modifier.component.ts (hunk 1 above)
Comment (LindvedKrvang) [reply src/app/settings/panel-settings/components/edit-panel-input-modifier/edit-panel-input-modifier.component.ts:48]: Good catch. It's been updated. Also updated some english values to danish as they were in the danish translation.

**Pull request merged**